py/dml/objects.py

```diff
@@ -461,8 +492,10 @@ def __init__(self, name, site, parent, msg_types, arraylens = ()):
         self.msg_types = msg_types
         self._arraylens = arraylens
         self.dimensions += self.local_dimensions()
-        self.uniq = len(dml.globals.hooks) + 1
-        dml.globals.hooks.append(self)
+        # during code generation, set to a unique positive integer
+        # (index in object.Device.hooks plus one, see
+        # CompositeObject.uniq)
+        self.uniq = None
```

Comment:
redundant

Reply:
No, hook is not a CompositeObject, and it has a `uniq` slot

Reply:
So it was, I failed at GitHub and thought it under `Group`.